01week/rockPaperScissors.js

```diff
@@ -8,9 +8,68 @@ const rl = readline.createInterface({
 });
 
 
+
 function rockPaperScissors(hand1, hand2) {
 
-  // Write code here
+  // create a sanitized array of the two hands for easier manipulation
+  let handArr = [hand1, hand2].map((x) => x.toLowerCase().replace(/\s/g, ''));
+
+  // declare an array of accepted hand values
+  const validHands = ['rock', 'paper', 'scissors', 'lizard', 'spock'];
+  
+  // return an error if the hands chosen are not in the array of accepted values
+  for (let value of handArr) {
+    if (!validHands.includes(value)) {
+      return 'Invalid input... Accepted values are ' + validHands.join(', ');
+    }
+  }
+
+  // if hand 1 and hand 2 are the same, it's a tie, we're done, return early
+  if (handArr[0] === handArr[1]) {
+    return 'It\'s a tie!';
+  }
+
+  // if it's not a tie, let's start by defining the scenarios where hand 1 wins
+  // let's also make a note of the verb that goes with each scenario
+  // then we can log an explanation to the console
+  const hand1Wins = [
+    {scenario: ['paper',    'rock'],     verb: 'covers'     },
+    {scenario: ['rock',     'scissors'], verb: 'crushes'    },
+    {scenario: ['scissors', 'paper'],    verb: 'cut'        },
+    {scenario: ['rock',     'lizard'],   verb: 'crushes'    },
+    {scenario: ['lizard',   'spock'],    verb: 'poisons'    },
+    {scenario: ['spock',    'scissors'], verb: 'smashes'    },
+    {scenario: ['scissors', 'lizard'],   verb: 'decapitate' },
+    {scenario: ['lizard',   'paper'],    verb: 'eats'       },
+    {scenario: ['paper',    'spock'],    verb: 'disproves'  },
+    {scenario: ['spock',    'rock'],     verb: 'vaporizes'  }
+  ];
+  
+
+  // the scenarios where hand 2 wins are going to be the same as hand 1
+  // but with rock/paper flipped to paper/rock, etc.
+  let hand2Wins = JSON.parse(JSON.stringify(hand1Wins));
```

Comment:
why are you stringifying hand1wins then parsing it back? That's really redundant